raft/src/main/java/org/apache/kafka/raft/MetadataLogConfig.java

```diff
@@ -16,25 +16,119 @@
+            "controller should write no-op records to the metadata partition. If the value is 0, no-op records " +
+            "are not appended to the metadata partition. The default value is " + METADATA_MAX_IDLE_INTERVAL_MS_DEFAULT;
+
+    public static final ConfigDef CONFIG_DEF =  new ConfigDef()
+            .define(METADATA_SNAPSHOT_MAX_NEW_RECORD_BYTES_CONFIG, LONG, METADATA_SNAPSHOT_MAX_NEW_RECORD_BYTES, atLeast(1), HIGH, METADATA_SNAPSHOT_MAX_NEW_RECORD_BYTES_DOC)
+            .define(METADATA_SNAPSHOT_MAX_INTERVAL_MS_CONFIG, LONG, METADATA_SNAPSHOT_MAX_INTERVAL_MS_DEFAULT, atLeast(0), HIGH, METADATA_SNAPSHOT_MAX_INTERVAL_MS_DOC)
+            .define(METADATA_LOG_DIR_CONFIG, STRING, null, null, HIGH, METADATA_LOG_DIR_DOC)
+            .define(METADATA_LOG_SEGMENT_BYTES_CONFIG, INT, METADATA_LOG_SEGMENT_BYTES_DEFAULT, atLeast(Records.LOG_OVERHEAD), HIGH, METADATA_LOG_SEGMENT_BYTES_DOC)
+            .defineInternal(METADATA_LOG_SEGMENT_MIN_BYTES_CONFIG, INT, METADATA_LOG_SEGMENT_MIN_BYTES_DEFAULT, atLeast(Records.LOG_OVERHEAD), HIGH, METADATA_LOG_SEGMENT_MIN_BYTES_DOC)
+            .define(METADATA_LOG_SEGMENT_MILLIS_CONFIG, LONG, METADATA_LOG_SEGMENT_MILLIS_DEFAULT, null, HIGH, METADATA_LOG_SEGMENT_MILLIS_DOC)
+            .define(METADATA_MAX_RETENTION_BYTES_CONFIG, LONG, METADATA_MAX_RETENTION_BYTES_DEFAULT, null, HIGH, METADATA_MAX_RETENTION_BYTES_DOC)
+            .define(METADATA_MAX_RETENTION_MILLIS_CONFIG, LONG, METADATA_MAX_RETENTION_MILLIS_DEFAULT, null, HIGH, METADATA_MAX_RETENTION_MILLIS_DOC)
+            .define(METADATA_MAX_IDLE_INTERVAL_MS_CONFIG, INT, METADATA_MAX_IDLE_INTERVAL_MS_DEFAULT, atLeast(0), LOW, METADATA_MAX_IDLE_INTERVAL_MS_DOC);
+
+    public final int logSegmentBytes;
+    public final int logSegmentMinBytes;
+    public final long logSegmentMillis;
+    public final long retentionMaxBytes;
+    public final long retentionMillis;
+    public final int maxBatchSizeInBytes;
+    public final int maxFetchSizeInBytes;
+    public final long deleteDelayMillis;
+    public final int nodeId;
+
+    /**
+     * Configuration for the metadata log
+     * @param logSegmentBytes The maximum size of a single metadata log file
+     * @param logSegmentMinBytes The minimum size of a single metadata log file
+     * @param logSegmentMillis The maximum time before a new metadata log file is rolled out
+     * @param retentionMaxBytes The size of the metadata log and snapshots before deleting old snapshots and log files
+     * @param retentionMillis The time to keep a metadata log file or snapshot before deleting it
+     * @param maxBatchSizeInBytes The largest record batch size allowed in the metadata log
+     * @param maxFetchSizeInBytes The maximum number of bytes to read when fetching from the metadata log
+     * @param deleteDelayMillis The amount of time to wait before deleting a file from the filesystem
+     * @param nodeId The node id
+     */
+    public MetadataLogConfig(int logSegmentBytes,
```

Comment:
could you please add a constructor accepting `AbstractConfig`? Also, please move the getters from `KafkaCOnfig` to this constructor. The getters are listed below.

```
config.metadataLogSegmentBytes,
        config.metadataLogSegmentMinBytes,
        config.metadataLogSegmentMillis,
        config.metadataRetentionBytes,
        config.metadataRetentionMillis,
```